nrf_modem/doc/CHANGELOG.rst

```diff
@@ -9,6 +9,41 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+nrf_modem 2.3.0
+***************
+
+Core library
+============
+* The :c:func:`nrf_modem_init` function is now only used to initialize the library in normal operating mode.
+  Use :c:func:`nrf_modem_bootloader_init` to initialize the library in bootloader mode.
+* Added a ``context`` parameter to :c:func:`nrf_modem_os_event_notify` to allow waking up only a subset of sleeping threads.
+* Added the :c:func:`nrf_modem_os_sleep` function.
+* The ``nrf_modem_limits.h`` file has been removed.
+
+Sockets
+=======
+* Added the ``NRF_SO_POLLCB`` socket option to receive callbacks for poll events occurring on a socket.
+* Added the :c:func:`nrf_getifaddrs` and :c:function:`nrf_freeifaddrs` functions to retrieve network interface data.
+* Fixed a bug where not reading incoming network data in a timely manner could hang the communication with the modem.
+* Fixed a bug in :c:func:`nrf_connect` where a blocking call could in certain cases time out and set the wrong ``errno`` (``EBUSY`` instead of ``ETIMEDOUT``).
+* Fixed a bug in :c:func:`nrf_poll` where only the first :c:struct:`nrf_pollfd` structure would be updated in case the modem was shutdown.
+* Fixed a bug in :c:func:`nrf_setsockopt` where setting ``NRF_SO_RAI_NO_DATA`` on a TCP socket where the peer had closed the connection would return an error.
+* Fixed a bug in :c:func:`nrf_send` and :c:func:`nrf_sendto` where the functions would hang when attempting to send a data payload larger than the TX region.
+* Fixed a possible concurrency bug in :c:func:`nrf_socket`.
+* Fixed a possible concurrency bug in :c:func:`nrf_accept`.
+
+AT interface
+============
+* Improved error checking in :c:func:`nrf_modem_at_cmd` and :c:func:`nrf_modem_at_printf`.
+
+GNSS interface
+==============
+* Added the :c:member:`nrf_modem_gnss_agps_expiry.position_expiry` field to :c:struct:`nrf_modem_gnss_agps_expiry` to retrieve the position assistance expiry time.
+
+Bootloader
+==========
+* The Full DFU API (:file:`nrf_modem_full_dfu.h`) has been moved to (:file:`nrf_modem_bootloader.h`) and renamed accordingly.
```

Comment:
Maybe add a line about prefixing the DFU results with `NRF_`? E.g. `NRF_MODEM_DFU_RESULT_OK`

Reply:
Also, some of the function signatures has changed the parameter order: (len, buf) -> (buf, len).